src/database/tuple_desc.cpp

```diff
@@ -1,36 +1,113 @@
 #include "tuple_desc.h"
 
 namespace emerald {
-TupleDesc::TupleDesc TupleDesc::combine(TupleDesc td1, TupleDesc td2) {
-}
-
 TupleDesc::TupleDesc() {
 }
 
-TupleDesc::TupleDesc(Type type_array[]) {
+TupleDesc::TupleDesc(std::vector<Type> type_array) {
```

Comment:
Sorry if this is kind of nitpicky, but a good practice if you want to name variables "type_array" is that if its type is really an array, it's good to then call it "type_array", but if it's a vector, it might be better to name it "type_vector". If you know what I mean!

We can discuss this if you like, just let me know what you think here!